airflow/providers/oracle/operators/oracle.py

```diff
@@ -63,4 +68,25 @@ def __init__(
     def execute(self, context) -> None:
         self.log.info('Executing: %s', self.sql)
         hook = OracleHook(oracle_conn_id=self.oracle_conn_id)
-        hook.run(self.sql, autocommit=self.autocommit, parameters=self.parameters)
+
+        def handler(cur):
+            bindvars = cur.bindvars
+
+            if isinstance(bindvars, list):
+                bindvars = [v.getvalue() for v in bindvars]
+            elif isinstance(bindvars, dict):
+                bindvars = {n: v.getvalue() for (n, v) in bindvars.items()}
+            else:
+                raise TypeError(bindvars)
+
+            return {
+                "bindvars": bindvars,
+                "rows": cur.fetchall(),
+            }
+
+        return hook.run(
```

Comment:
There is a subtle problem with this approach. This would mean that the Oracle Provider will be only compatible with the new DBApi implementation, which is part of the Airflow itself. This means that either we make a breaking change (and add > 2.1 limitation to the Oracle provider, or (better) we implement it in backward-compatible way - honoring the fact that the run method might have no "handler" added.

That would mean that bind-variable functionality will only be available in Airflow 2.1 and above

Reply:
We could do in one of the ways described here: https://stackoverflow.com/questions/11915032/get-keyword-arguments-for-function-python

Reply:
And I have to think about automating such checks in pre-commit (mental note to self).

Reply:
@potiuk to some extent I feel that this is a big enough change either way that perhaps a 2.1 target is not so bad – in how it'll actually fetch all rows now and – given that `do_push_xcom` is set by default – pushed to Xcom.

Reply:
The problem with that is that we are releasing providers from "master"  so if we make a "2.1" change, it will have to wait until to 2.1 release to be merged. 

IMHO much better solution i  implementing it in a backwards-compatible version,  merging it now and releasing new provider that will simply use the functionality when Airflow 2.1 is released.

Reply:
@potiuk good point – fixed now in b81bf4d6de52d4528e58324722a2118235570da0.

Thanks!